tests/test_internals/test_si.py

```diff
@@ -576,5 +579,74 @@ def test_group_set_custom_description(self):
 
 
+class TestSpikeInterfaceRecorders(unittest.TestCase):
+    def setUp(self):
+
+        self.test_dir = tempfile.mkdtemp()
+        self.path1 = self.test_dir + "/test_electrodes1.nwb"
+
+        self.nwbfile1 = NWBFile("session_description1", "file_id1", datetime.now())
+
+        self.RX1 = generate_recording(num_channels=4, durations=[3])
+        self.metadata_list = [dict(Ecephys={i: dict(name=i, description="desc")}) for i in ["es1", "es2"]]
+
+    def tearDown(self):
+        del self.RX1
+        shutil.rmtree(self.test_dir)
+
+    def test_non_ints_as_channel_ids(self):
+
+        channel_ids = self.RX1.get_channel_ids()
+        num_channels = self.RX1.get_num_channels()
+        non_int_channel_ids1 = ["a", "b", "c", "d"]
+        non_int_channel_ids2 = ["c", "d", "e", "f"]
+
+        self.RX_non_int_channels1 = self.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids1
+        )
+        self.RX_non_int_channels2 = self.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids2
+        )
+
+        values_before_rewrite = ["value_before_rewrite" for _ in range(num_channels)]
+        self.RX_non_int_channels1.set_property("property", values=values_before_rewrite)
+
+        values_after_rewrite = ["value_after_rewrite" for _ in range(num_channels)]
+        self.RX_non_int_channels2.set_property("property", values=values_after_rewrite)
+
+        # Add a property only for two
+        property_value = ["second_recoder_property" for _ in range(num_channels)]
+        self.RX_non_int_channels2.set_property("property2", values=property_value)
+
+        # Write
+        write_recording(
+            recording=self.RX_non_int_channels1, nwbfile=self.nwbfile1, metadata=self.metadata_list[0], es_key="es1"
+        )
+        # Re-write
+        write_recording(
+            recording=self.RX_non_int_channels2, nwbfile=self.nwbfile1, metadata=self.metadata_list[1], es_key="es2"
+        )
+
+        with NWBHDF5IO(str(self.path1), "w") as io:
+            io.write(self.nwbfile1)
+        with NWBHDF5IO(str(self.path1), "r") as io:
+            nwb = io.read()
+
+            # First we test the channel names are assigned in the written order
+            expected_channel_names = ["a", "b", "c", "d", "e", "f"]
+            for id in nwb.electrodes.id[:]:
+                assert nwb.electrodes["channel_name"][id] == expected_channel_names[id]
+
+            # Test extension of the same property for the new channels
+            for id in [0, 1, 2, 3]:
+                assert nwb.electrodes["property"][id] == "value_before_rewrite"
+            for id in [4, 5]:
+                assert nwb.electrodes["property"][id] == "value_after_rewrite"
+
+            # Test addition of new property for the second recording
+            for id in [2, 3, 4, 5]:
+                assert nwb.electrodes["property2"][id] == "second_recoder_property"
```

Comment:
(c) Yet another  habit to get into would be making a separate test function for each case and naming the functions appropriately to the purpose of the test. So each of these parts indicated by comments could be their own tests, and the common parts of the `setUp` could be propagated into the actual class setup.

Reply:
I gave this a thought and I am not sure what do we gain. I think that all these test make sense together as they test writing non int channel ids to the electrode table. 

If I were to partition there would be three functions instead of one with the following names:
* test_non_ints_as_channel_ids_in_electrode_table_channel_name.
* test_non_ints_as_channel_ids_in_electrode_table_property_extension.
* test_non_ints_as_channel_ids_in_electrode_table_new_property_addition.

This increases a little bit the boilerplate and the future user of the test needs to jump back and forth between the actual test and the `setUp` definition which is not too bad here as both things are small but what do we gain? this is not clear to me.

Reply:
This makes sense.